sympy/core/mul.py

```diff
@@ -86,12 +86,60 @@ def _unevaluated_Mul(*args):
 
 
 class Mul(Expr, AssocOp):
+    """
+    Multiplication operation for algebraic group.
+
+    This class mainly deals with the multiplication in the field of complex numbers.
+    Any abstract algebraic field can be dealt as well, but it is encouraged to
+    define specific class such as ``MatMul`` or ``VectorMul`` for other fields.
+    This class also serves as base class for these classes.
+
+    Another use of ``Mul()`` is to behave as template for abstract multiplication so
+    that its arguments can be substituted to return different class. Refer to
+    examples section for this.
+
+    Since multiplication can be vector space operation, arguments may have the different
+    kind. Kind of the resulting object is automatically inferred.
 
+    Examples
+    ========
+
+    >>> from sympy import Mul
+    >>> from sympy.abc import x, y
+    >>> Mul(x, 1)
+    x
+    >>> Mul(x, x)
+    x**2
+
+    If ``evaluate=False`` is passed, result is not evaluated.
+
+    >>> Mul(1, 2, evaluate=False)
+    1*2
+    >>> Mul(x, x, evaluate=False)
+    x*x
+
+    ``Mul()`` also serves as a template for other classes.
+
+    >>> from sympy import MatrixSymbol
+    >>> A = MatrixSymbol('A', 2,2)
+    >>> expr = Mul(x,y).subs({y:A})
+    >>> expr
+    x*A
+    >>> expr.func
+    <class 'sympy.matrices.expressions.matmul.MatMul'>
```

Comment:
I think it would be better to use `type` rather than `func`.

In general `func` should not be presumed to be the same as `type` and should be understood only as a callable that can be used as `obj.func(*obj.args)`.